### PR TITLE
T287146: language selector

### PR DESCRIPTION
**Pull request by medied**
https://phabricator.wikimedia.org/T287146

### Objective

Adds language selector feature

**TODO** 

- [x] keyboard controllers?
- [x] hover styling (cursor and background color)
- [x] 🐛 if you open the language selector with empty title, gotta set the height somehow. 
- [x] all languages label
- [x] language-data decision

### CODE REVIEW

File: src/link/style.scss [hunk 1]
```diff
@@ -131,6 +174,97 @@
...
+			margin-left: 16px;
+		}
+
+		&.no-title {
+			position: relative;
+			top: -120px;
+			height: 310px;
+		}
+	}
+
```
Comment (hueitan) [inline src/link/style.scss:267]: missing support of `RTL` interface

File: src/link/language-selector.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	return (
+		<div
+			className={ `wikipediapreview-edit-inline-language-selector ${
+				! title ? `no-title` : ''
+			}` }
+		>
+			<div className="wikipediapreview-edit-inline-language-selector-header">
+				<div>{ __( 'Languages', 'wikipedia-preview' ) }</div>
+				<div
+					className="wikipediapreview-edit-inline-language-selector-header-close"
```
Comment (hueitan) [inline src/link/language-selector.js:82]: Getting two X in the mobile interfac

File: src/link/language-selector.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+			return;
+		}
+
+		const filtered = Object.keys( languages )
+			.filter( ( language ) => {
+				if ( languages[ language ].length > 2 ) {
+					return (
+						languages[ language ][ 2 ]
+							.toLowerCase()
+							.indexOf( targetLang.toLowerCase() ) !== -1
```
[the remark this answers is not in the document]
Comment (medied) [reply src/link/language-selector.js:38]: Not a bad idea as it would make the search more robust

[the remark this answers is not in the document]
Comment (hueitan) [reply src/link/language-selector.js]: it would be nice also trim the targetLang

File: src/link/style.scss [hunk 4]
```diff
@@ -37,13 +41,51 @@
...
+		&-language {
+			position: absolute;
+			bottom: 0;
+			right: 10px;
+			display: flex;
+			width: 50px;
+			justify-content: space-evenly;
+			align-items: center;
+			border-left: solid 1px;
+			height: 30px;
```
Comment (hueitan) [inline src/link/style.scss:53]: missing `cursor: pointer`

File: src/link/style.scss [hunk 5]
```diff
@@ -37,13 +41,51 @@
 			}
 		}
 
+		&-language {
```
Comment (hueitan) [inline src/link/style.scss:44]: I think the box need more space on the left, it's now so close to the line border;

| image | image |
| --- | --- |

another styling icon, it looks offset now (same to mobile header)

| image | image |
| --- | --- |

**Pull request merged**